tiledb/query_condition.py

```diff
@@ -0,0 +1,149 @@
+import ast
+
+import tiledb
+from tiledb import _query_condition as qc
+
+"""
+A high level wrapper around the Pybind11 query_condition.cc implementation for
+filtering query results on attribute values.
+"""
+
+
+class QueryCondition(ast.NodeVisitor):
+    """
+    Class representing a TileDB query condition object for attribute filtering
+    pushdown. Set the query condition with a string representing an expression
+    as defined by the grammar below. A more straight forward example of usage is
+    given beneath.
+
+    BNF
+    ---
+    An expression is made up of one or more Boolean expressions. Multiple Boolean
```

Comment:
Use term "statement" instead of "expression" to clearly differentiate from "Boolean expression"